From 5e87ff3c8337cdf073e631e1257efe4f8f6ecbcf Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Wed, 7 Jun 2023 10:47:48 +0200
Subject: [PATCH 64/95] arm64: dts: st: add GPU on stm32mp25

Add GPU support for stm32mp25

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git e04eb0c3e8005bb13b19f104a8d96a5d6ed5dd0d arm64: dts: st: add GPU on stm32mp25f-ev1 daughter board
ff202bc2945704d744a31f8e7b76c6ee07d413e3 arm64: dts: st: add GPU on stm32mp25f-dk daughter board
86872bf3882af1f090d4bbeb3eb5f324416acde6 arm64: dts: st: add GPU on stm32mp257f-ev1 board
459c08e245c42cd5769c13fc8a8813ab12cc93fa arm64: dts: st: add GPU on stm32mp25]
Signed-off-by: Pierre-Yves MORDRET <pierre-yves.mordret@foss.st.com>
Change-Id: Ib901afe556d318f40cd2e7c8064fc1bbcf7206a7
---
 arch/arm64/boot/dts/st/stm32mp255.dtsi | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp255.dtsi b/arch/arm64/boot/dts/st/stm32mp255.dtsi
index 7a598f53a2a0e..200adc8625ad2 100644
--- a/arch/arm64/boot/dts/st/stm32mp255.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp255.dtsi
@@ -40,4 +40,16 @@ venc: venc@480e0000 {
 		clocks = <&rcc CK_BUS_VENC>;
 		access-controllers = <&rifsc 90>;
 	};
-};
\ No newline at end of file
+
+	gpu: gpu@48280000 {
+		compatible = "vivante,gc";
+		reg = <0x48280000 0x800>;
+		interrupts = <GIC_SPI 183 IRQ_TYPE_LEVEL_HIGH>;
+		resets = <&rcc GPU_R>;
+		clock-names = "axi", "core";
+		clocks = <&rcc CK_BUS_GPU>, <&rcc CK_KER_GPU>;
+		gpu-supply =  <&scmi_vddgpu>;
+		access-controllers = <&rifsc 79>;
+		status = "disabled";
+	};
+};
-- 
2.51.0

